exovetter/vetters.py

```diff
@@ -137,10 +124,107 @@ def plot(self):
                 'LPP plot data is empty. Execute self.run(...) first.')
 
         # target is populated in TCE, assume it already exists.
-        target = self.tce['target_name']
+        target = self.tce.target_name
         lpp.plot_lpp_diagnostic(self.plot_data, target, self.norm_lpp)
 
 
+class Sweet(BaseVetter):
+    """Class to handle SWEET Vetter functionality.
+
+    Parameters
+    ----------
+    threshold_sigma : float
+        Threshold for comparing signal to transit period.
+
+    Attributes
+    ----------
+    tce, lc
+        Inputs to :meth:`run`.
+
+    result : dict
+        ``'amp'`` contains the best fit amplitude, its uncertainty, and
+        amplitude-to-uncertainty ratio for half-period, period, and
+        twice the period. ``'msg'`` contains warnings, if applicable.
+        Populated by :meth:`run`.
+
+    lsf : `~exovetter.utils.WqedLSF`
+        Least squares fit object, populated by :meth:`run`.
+
+    """
+    def __init__(self, threshold_sigma=3):
+        self.tce = None
+        self.lc = None
+        self.result = None
+        self.lsf = None
+        self.threshold_sigma = threshold_sigma
+
+    def run(self, tce, lightcurve):
+        self.tce = tce
+        self.lc = lightcurve
+        self.result, self.lsf = self._do_fit(
+            lightcurve.time, lightcurve.flux,
+            tce.period, tce.tzero, tce.duration)
+
+    def plot(self):
```

Comment:
Is this plot what you want? Or is it something else? See posted screenshot above.

Reply:
The plot would be more useful if you could see the transit and ellipsoidal variations (the red line) with greater range in y. Divide the transit by the mean, or use subplots.

Reply:
I updated the plot above. Better now?

Reply:
Yes, this plot looks good.